File: Week_09/min_cost_climbing_stairs.js
var minCostClimbingStairs = function(cost) {
  cost.push(0);
  let n = cost.length;
  let dp = [];
  dp[0] = cost[0]; 
  dp[1] = cost[1]; 
  for(let i = 2;i < n;i++){
      dp[i] = Math.min(dp[i-2] , dp[i-1]) + cost[i];
  }
  return dp[n-1];
};

File: Week_09/min_cost_climbing_stairs.go
func minCostClimbingStairs(cost []int) int {
	var i int
	if len(cost) == 0 {
			return 0
	}
	if len(cost) == 1{
			return cost[0]
	}
	for i = len(cost) - 3 ; i >= 0; i-- {
			cost[i] += min(cost[i + 1],cost[i + 2])
	}
	return min(cost[1] , cost[0])
}
func min(a, b int) int{
	if a > b {
			return b
	}else{
			return a
	}
}
